CommonTools/Utils/interface/KinematicColumns.h

```diff
@@ -9,13 +9,30 @@ namespace edm {
 
     namespace col {
 
-      SOA_DECLARE_COLUMN(Pt, float, "pt");
       SOA_DECLARE_COLUMN(Eta, float, "eta");
       SOA_DECLARE_COLUMN(Phi, float, "phi");
+      SOA_DECLARE_COLUMN(Theta, float, "theta");
+
       SOA_DECLARE_COLUMN(Vz, float, "vz");
 
+      SOA_DECLARE_COLUMN(Px, float, "px");
+      SOA_DECLARE_COLUMN(Py, float, "py");
+      SOA_DECLARE_COLUMN(Pz, float, "pz");
+      SOA_DECLARE_COLUMN(Pt, float, "pt");
+      SOA_DECLARE_COLUMN(P, float, "p");
+
     }  // namespace col
 
+    SOA_DECLARE_DEFAULT(col::Eta, eta());
+    SOA_DECLARE_DEFAULT(col::Phi, phi());
+    SOA_DECLARE_DEFAULT(col::Theta, theta());
+    SOA_DECLARE_DEFAULT(col::Vz, vz());
+    SOA_DECLARE_DEFAULT(col::Px, px());
+    SOA_DECLARE_DEFAULT(col::Py, py());
+    SOA_DECLARE_DEFAULT(col::Pz, pz());
+    SOA_DECLARE_DEFAULT(col::Pt, pt());
+    SOA_DECLARE_DEFAULT(col::P, p());
```

Comment:
Templated `value_for_column()` definitions would better be placed in the global namespace. Placing them in `edm::soa` effectively prevents providing an overload in a global namespace for a type `T` in a namespace different from `edm::soa` or global.

Reply:
In general the `value_for_column()` overloads are best to be placed in the same namespace as the source type (similar to `std::swap()`. I've clarified this point in the documentation.

Reply:
Hi, sorry I'm kind of blocked here. If I put the `SOA_DECLARE_DEFAULT` calls in the global namespace, like:
```C++
...
SOA_DECLARE_DEFAULT(edm::soa::col::Py, py());
...
```
(also for TrackSpecificColumns.h), I get compiler errors which I really could not figure out even after taking quite some time to ponder. For some reason, it does not find the `value_for_column` implementations for the track specific columns anymore when it tries to create the `egamma::conv::TrackTable` in GsfElectronAlgo. I have to come back to this later when I hopefully have fresh ideas, but for the moment I really have no clue.

Reply:
After playing around a bit with ADL I think

> Templated `value_for_column()` definitions would better be placed in the global namespace. 

I was wrong with this suggestion (sorry for that), and the templated defaults should be in `edm::soa` namespace as they originally were. Then the order of `#include`s should not matter (the caller of unqualified `value_for_filler()` appears to always find the templated overload in `edm::soa` regardless of the order of the function declarations).

> Placing them in `edm::soa` effectively prevents providing an overload in a global namespace for a type `T` in a namespace different from `edm::soa` or global.

Essentially I got blinded by assuming placing the non-templated overloads to global namespace would be the way to go. Instead, they would be best placed in the namespace of the object being filled from (technically `edm::soa` appears to work similarly).

Reply:
Hi, thanks for your investigation! Very nice, then I can bring the templated `value_for_column()` back into the `edm::soa` namespace and address your remaining comment.